src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantCallerUtils.java

```diff
@@ -38,25 +39,106 @@ static int getTotalHardClipping(final Cigar cigar) {
+     * of the list ({@code fromStart==true}) or from the end of the list ({@code fromStart==false}).
+     *
+     * @throws IllegalArgumentException if fails check by {@link #checkCigarElements(List)}
+     */
+    @VisibleForTesting
+    static int getNumHardClippingBases(final boolean fromStart, final List<CigarElement> cigarElements) {
+
+        checkCigarElements(cigarElements);
+
+        // "H can only be present as the first and/or last operation" according to VCF spec 4.2
+        final int index = fromStart ? 0 : cigarElements.size()-1;
+        final CigarElement firstElement = cigarElements.get(index);
+        return firstElement.getOperator()== CigarOperator.H ? firstElement.getLength() : 0;
+    }
+
+    /**
+     * @return the number of soft clipped bases as indicated in the input {@code cigarElements}, either from the beginning
+     * of the list ({@code fromStart==true}) or from the end of the list ({@code fromStart==false}).
+     *
+     * @throws IllegalArgumentException if fails check by {@link #checkCigarElements(List)}
+     */
+    @VisibleForTesting
+    static int getNumSoftClippingBases(final boolean fromStart, final List<CigarElement> cigarElements) {
+
+        checkCigarElements(cigarElements);
+
+        // because 'H' can only be the 1st/last operation according to the spec, and also
+        // "S may only have H operations between them and the ends of the CIGAR string",
+        // 'S' could only be the 1st operation or the 2nd operation next to the 'H' sitting at the end
+        // no two 'S' operations should sit next to each other
+
+        final int endIndex = fromStart ? 0 : cigarElements.size()-1;
+        CigarElement element = cigarElements.get(endIndex);
+        if (element.getOperator().isClipping()) {
+            if (element.getOperator()==CigarOperator.S) {
+                return element.getLength();
+            } else {
+                final CigarElement mayBeSoftClipping = cigarElements.get(fromStart ? 1 : cigarElements.size()-2);
+                return mayBeSoftClipping.getOperator()==CigarOperator.S ? mayBeSoftClipping.getLength() : 0;
+            }
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * Checks input list of cigar operations for:
+     * <ul>
+     *     <li>empty input list;</li>
+     *     <li>when having only 1 operations, it must be an alignment;</li>
+     *     <li>deletion operation cannot neighbor clipping operations;</li>
+     *     <li>there must be at least one alignment operation in the list</li>
+     * </ul>
+     * @param cigarElements
+     */
+    @VisibleForTesting
+    static void checkCigarElements(final List<CigarElement> cigarElements) {
+        Utils.validateArg(!cigarElements.isEmpty(), "Cannot parse empty list cigarElements");
+        Utils.validateArg(cigarElements.size()>1 || (cigarElements.size()==1 && cigarElements.get(0).getOperator().isAlignment()),
```

Comment:
Yes, you are right in that the 4th check covers the 2nd and the 2nd is not a good check. 
The case of `10H10S1D10M` is now caught as well, with new test coverage. Thanks!